p2p/README.md

```diff
@@ -67,105 +66,431 @@ flowchart TD
+        end
+
+        rp2ph[[`PocketEnvelope` Handler]]
+        rbus[bus]
+        rhost -. "new stream" .-> rrtu
+        rhost -- "new subscription message" --> rbgu
+        rrtu --> rrth
+
+        rnd[Nonce Deduper]
+        rp2ph -- "deduplicate msg mempool" --> rnd
     end
-    subgraph lRT[Raintree Router]
+
+
+    rp2ph -. "(iff not duplicate msg)\npublish event" .-> rbus
+
+    rrth --> rp2ph
+    rbgh --> rp2ph
+
+    lhost --> rhost
+```
+
+**Broadcast**
+
+```mermaid
+flowchart
+  subgraph lp2p["Local P2P Module (outgoing)"]
+    lpb[[`Broadcast`]]
+    lpb -. "(iff local & remote peer are staked)" ..-> lrtu
+    lpb -- "(always)" --> lbggt
+    
+    lbggt -- "msg published\n(gossipsub protocol)" ---> lhost
+    
+    lhost[Libp2p Host]
+
+    subgraph lrt[RainTree Router]
       subgraph lRTPS[Raintree Peerstore]
         lStakedPS([staked actors only])
       end
-
-      subgraph lPM[PeerManager]
-      end
-      lPM --> lRTPS
+      
+      lrtu[UnicastRouter]
+      
+      lrtu -- "network address lookup" --> lRTPS
     end
+    
+    lrtu -- "opens a stream\nto target peer" ---> lhost
 
-    subgraph lBG[Background Router]
+    subgraph lbg[Background Router]
+      lbggt[Gossipsub Topic]
       subgraph lBGPS[Background Peerstore]
         lNetPS([all P2P participants])
       end
+      
+      lbggt -- "network address lookup" --> lBGPS
+    end
+  end
 
-      subgraph lGossipSub[GossipSub]
-      end
+  subgraph rp2p["Remote P2P Module (incoming)"]
+    rhost[Libp2p Host]
 
-      subgraph lDHT[Kademlia DHT]
-      end
+    subgraph rrt[RainTree Router]
+      rrth[[`RainTreeMessage` Handler]]
+      rrtu[UnicastRouter]
+    end
 
-      lGossipSub --> lBGPS
-      lDHT --> lBGPS
+    subgraph rbg[Background Router]
+      rbgh[[`BackgroundMessage` Handler]]
+      rbgg[Gossipsub Subscription]
+      rbggt[Gossipsub Topic]
+      rbgg --> rbgh
+      rbgh -- "(background msg\npropagation cont.)" ---> rbggt
     end
 
-    lRT --1a--> lHost
-    lBG --1b--> lHost
+    rp2ph[[`PocketEnvelope` Handler]]
+    rbus[bus]
+    rhost -. "new stream" ..-> rrtu
+    rhost -- "new subscription message" --> rbgg
+    rbggt -- "(background msg\npropagation cont.)" --> rhost
+    rrtu --> rrth
+    rrth -. "(iff level > 0)\n(raintree msg\npropagation cont.)" .-> rrtu
+    rrtu -- "(raintree msg\npropagation cont.)" --> rhost
+
+    rnd[Nonce Deduper]
+    rp2ph -- "deduplicate msg mempool" --> rnd
   end
 
-subgraph rMod[Remote P2P Module]
-subgraph rHost[Libp2p `Host`]
-end
-subgraph rRT[Raintree Router]
-subgraph rPS[Raintree Peerstore]
-rStakedPS([staked actors only])
-end
 
-subgraph rPM[PeerManager]
-end
+  rp2ph -. "(iff not duplicate msg)\npublish event" .-> rbus
 
-rPM --> rStakedPS
-end
+  rrth --> rp2ph
+  rbgh --> rp2ph
 
-subgraph rBG[Background Router]
-subgraph rBGPS[Background Peerstore]
-rNetPS([all P2P participants])
-end
+  lhost --> rhost
+```
 
-subgraph rGossipSub[GossipSub]
-end
+### Message Deduplication
```

Comment:
Can you add a note on how/when the deduplication is cleared? On node restart? Configuration? etcc